test/powershell/Language/LanguageTestSupport.psm1

```diff
@@ -24,18 +24,57 @@ function Get-RuntimeError
     [CmdletBinding()]
     param(
         [Parameter(ValueFromPipeline=$True,Mandatory=$True)]
-        [string]$src
+        [string]$src,
+        [Parameter()]
+        [int]$Timeout = 0
     )
-
+ 
     $errors = $null
     try 
     {
-        [scriptblock]::Create($src).Invoke() > $null
+        # some tests cannot be run in a isolated runspace
+        # easily because they require more than just a single
+        # execution
+        if ( $Timeout -eq 0 )
+        {
+            [scriptblock]::Create($src).Invoke() > $null
+        }
+        else
+        {
+            $ps = [powershell]::Create()
+            $ps.AddScript($src) > $null
+            $ar = $ps.BeginInvoke()
+            # give it 250 milliseconds to complete
+            start-sleep -mill 250
+            if ( $ar.IsCompleted ) {
+                $ps.EndInvoke($ar)
+            }
+            # wait another ${Timeout} seconds, then give up
+            else {
+                Start-Sleep -sec $Timeout
+                if ( $ar.IsCompleted ) {
+                    # this can throw, which will be picked up below
+                    $ps.EndInvoke($ar)
+                }
+                else {
+                    # if it didn't throw, then return a constructed error
+                    $ER = Write-Error "Operation Timed Out ('$src')" 2>&1
+                    return $ER
+                }
+            }
+        }
     }
     catch 
     {
+        write-verbose -verbose "caught"

```

Comment:
indeed yes - pulled

---
In reply to: [94878328](https://github.com/PowerShell/PowerShell/pull/2958#discussion_r94878328) [](ancestors = 94878328)